utilities/speedrank-utils.js

```diff
@@ -0,0 +1,304 @@
+              console.log(err);
+              msg.channel.send(
+                'The bot user has not set up valid Google API credentials yet.')
+              .then((res) => {
+                resolve(res);
+                return;
+              });
+            }
+          } else {
+            // CategoryNames (headers) are fixed for the rank request
+            let categoryNames = ['Boss', 'Rank', 'Time'];
+
+            //Fight names are always in Row 2
+            const fightRow = data.values[2];
+
+            let fightNames = [];
+            for(let i = 1; i < fightRow.length; i++) {
+              if (fightRow[i] === undefined ||
+                fightRow[i] === '' ||
+                fightRow[i] === null) {
+                continue;
+              }
+              else {
+                fightNames.push(fightRow[i]);
+              } 
+            }
+
+
+            let contestants = [];
+            let padLength = 0;
+
+            // Now that we know the fights, let's search each fight column for the player in question
+           for (let catName of fightNames) {
+              let categoryRange = find(catName, data.values);
+
+              //TODO: Get max rows of table
+              for (let i = 0; i < 500; i++) {
+                // categoryRange.row + 2 will give us the starting row of
+                // contestants.
+                let row = categoryRange.row + 2 + i;
+                //console.log("row = " + row);
+                // Stop processing if we've hit the end of the list.
+                if (data.values[row] === undefined) {
+                  break;
+                }
+                let contestant = [];
+                // entryStartPos gives us the starting cell
+                // with the contestant name.
+                const entryStartPos = categoryRange.columnNum - 1;
+                let cell = checkCell(data.values[row][entryStartPos]);
+                if (catName.length > padLength) {
+                  padLength = catName.length + 1;
+                  console.log(padLength);
+                }
+                if (cell.toLowerCase() === player.toLowerCase()) {
+                  contestant.push(catName);
+                  // Grab the actual rank of this row
+                  contestant.push(checkCell(data.values[row][0]));
+
+                  // The Overall category and Torment sheets places avg time in the third column. Others have them in the second
+                  if (catName === "Overall" || categorySheet === 'Torment'){
+                    contestant.push(checkCell(data.values[row][entryStartPos+2]));
+                  }
+                  else {
+                    contestant.push(checkCell(data.values[row][entryStartPos+1]));
+                  }
+
+                  contestants.push(contestant);
+                  break;
+                }
+              }
+            }
+            const rankTable =
+              outputRankTable(categorySheet, player,
```

Comment:
Since you're using `player` for your `outputRankTable`, this will use the user's input exactly as requested in the final output. Everything looks right, for example, if you use `rank EverythingIsGravy`, things look fine since that's how they format their own name on the spreadsheet, but if you do `rank EVERYTHINGISGRAVy`, then the results will show:

```
EVERYTHINGISGRAVy - GL 4* Overall rankings
Boss          | Rank | Time  | 
------------------------------
Maliris       | 2    | 7.80  | 
Isgebind      | 2    | 8.02  | 
Tiamat        | 3    | 8.79  | 
Midgardsormr  | 11   | 11.20 | 
Ixion         | 3    | 4.79  | 
Kraken        | 5    | 9.06  | 
Siren         | 5    | 9.27  | 
Hades         | 10   | 13.37 | 
Overall       | 8    | 9.04  | 
```

In line 119, you do a comparison of `cell.toLowerCase` with the player's `toLowerCase`. If it matches, you might want to put the result of `cell.toLowerCase` in its own separate variable, say `playerActualName`, and use that as your input in `outputRankTable`. That way, no matter what the user inputs, the output will always have the correct way the name is displayed on the sheet.

Reply:
I did this partially as requested. Rather than use `cell.toLowerCase` I opted for just `cell` as it preserves the formatting the player used for their initial submission to the sheet.

As you suggested above I will break the redundant functions out in a separate commit.